rasterio/_warp.pyx

```diff
@@ -361,10 +387,14 @@ def _reproject(
     cdef GDALTransformerFunc pfnTransformer = NULL
     cdef GDALWarpOptions *psWOptions = NULL
 
+    # Set up GDALCreateGenImgProjTransformer2 keyword arguments.
+    cdef char **imgProjOptions = NULL
+    CSLSetNameValue(imgProjOptions, "GCPS_OK", "TRUE")
```

Comment:
Currently, GDAL will choose a polynomial order for you based on the number of GCPs. There are indeed options to override, to use TPS, and to filter GCP outliers. These will require that we call `GDALCreateGenImgProjTransformer2()` (as you've pointed out, @perrygeo). I'll make sure to do that, but let's leave rio-warp command line options to another PR.

Will add `id` and `description` to `GroundControlPoint`. I think I'll probably also make another PR that lets you dump a dataset's GCPs as GeoJSON. `rio info --gcps`?